### PR TITLE
feat: add tests for the /planters/{planterId} page

### LINKED ISSUE #132: The /planters/{planterId} page
* Implement the /planters/{id} page.
* Write integration test to cover the page, check the [guide](https://github.com/Greenstand/treetracker-web-map-client#test) of int test. 
* The UI: https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap?node-id=2497%3A11031
* Prerequites: a common component in this page, check video below.

https://www.loom.com/share/547c1e48753648739ad0dc7a376ded65

image

---
Some hints:
* Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
* Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
* To know more about our organization, visit our [website](https://greenstand.org).

### PR DESCRIPTION
**Pull request by pierrelstan**
# Description
The following changes have been implemented:

-Test integration for the planters page

Fixes #132
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by QuaidBartolomei** [comment]
@pierrelstan Automated workflow testing was recently added to the web-map-site branch. Rebase this branch with the new changes and we can see it in action 😃

**Comment by pierrelstan** [comment]
> @pierrelstan Automated workflow testing was recently added to the web-map-site branch. Rebase this branch with the new changes and we can see it in action 😃

@QuaidBartolomei   I updated the branch.

**Comment by QuaidBartolomei** [comment]
@pierrelstan The test does not pass with nock (npm run cy:nock). The planterid page is making multiple api calls in its SSR function, and each one needs to be mocked.

`[planterid].js`
```js 
    const { featured_trees, associated_organizations, species } =
      props.planter.links;
    props.planter.featuredTrees = await utils.requestAPI(featured_trees);
    props.planter.associatedOrganizations = await utils.requestAPI(
      associated_organizations,
    );
    props.planter.species = await utils.requestAPI(species);
  }
```

featured_trees, associated_organizations, and species all need to be mocked as well. See `top.cy.js` for an example of mocking multiple routes with nock. 

`top.cy.js`
```jsx
      cy.task('nocks', {
        hostname: 'http://127.0.0.1:4010/mock',
        routes: [
          {
            method: 'GET',
            path: '/trees/featured',
            statusCode: 200,
            body: {
              trees: [{ ...tree186734 }],
            },
          },
          {
            method: 'GET',
            path: '/trees/186734',
            statusCode: 200,
            body: { ...tree186734 },
          },
          {
            method: 'GET',
            path: '/countries/leader',
            statusCode: 200,
            body: leader,
          },
        ],
      });
```

**Comment by pierrelstan** [comment]
@QuaidBartolomei,  I have added the changes recommended.

File: cypress/tests/integration/planters/[planterid].cy.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  it('should render planters/{planterId} page and display the name of the planter', () => {
+    const path = `/planters/${planter.id}`;
+
+    cy.intercept(path, { doc: { example: '940.json' } });
+
+    cy.request(path);
+
+    cy.get('.MuiTypography-h2').contains(/sebastian gaertner/i);
+  });
```
Comment (QuaidBartolomei) [inline cypress/tests/integration/planters/[planterid].cy.js:91]: This test is not passing with `npm run cy:nock`. It seems a redundant with the first test though. I think this test can be removed.

` cy.get('.MuiTypography-h2').contains(/sebastian gaertner/i);`
This assertion can be used in the first test to improve it.

**Review by QuaidBartolomei — CHANGES_REQUESTED** [review]
Nice job mocking the other requests 👍. The tests are still not passing though, see the other comment for details. It is important that all of the tests are able to pass so that they can eventually be used for things like the github pr workflow.

File: cypress/tests/integration/planters/[planterid].cy.js (hunk 1 above)
Comment (pierrelstan) [reply cypress/tests/integration/planters/[planterid].cy.js]: @QuaidBartolomei, I removed it, I didn't notice that. Thank you.

**Review by QuaidBartolomei — APPROVED** [review]
Good job with the fixes 🥇

**Comment by pierrelstan** [comment]
> Good job with the fixes 🥇

Thank you

**Comment by dadiorchen** [comment]
@pierrelstan thanks, great job!

**Pull request merged**